app/helpers/application_helper.rb

```diff
@@ -123,6 +123,8 @@ def follow_button(followable, style = "full", classes = "")
     return if followable == DELETED_USER
 
     user_follow = followable.instance_of?(User) ? "follow-user" : ""
+    followable_type = followable.decorated? ? followable.object.class.name.downcase : followable.class.name.downcase
```

Comment:
Nitpick: We can save some repetition here:

```suggestion
    (followable.decorated? ? followable.object : followable).class.name.downcase
```